lib/NodeConnection.js

```diff
@@ -179,41 +189,42 @@ define(function (require, exports, module) {
      * Helper function to attempt a single connection to the node server
      */
     NodeConnection.prototype._attemptSingleConnect = function () {
-        var deferred = Promise.defer();
-        var port = null;
-        var ws = null;
-        setDeferredTimeout(deferred, CONNECTION_TIMEOUT);
+        var getRemotePromised = Promise.promisify(this.getRemotePort, this);
 
-        this.getRemotePort(function (err, nodePort) {
-            if (err) {
-                deferred.reject(err);
-                return;
-            }
+        var _connect = function (nodePort) {
+            return new Promise(function (success, failure) {
+                var port = nodePort;
 
-            port = nodePort;
-            ws = new WebSocket("ws://localhost:" + port);
+                var ws = new WebSocket("ws://localhost:" + port);
 
-            // Expect ArrayBuffer objects from Node when receiving binary
-            // data instead of DOM Blobs, which are the default.
-            ws.binaryType = "arraybuffer";
+                // Expect ArrayBuffer objects from Node when receiving binary
+                // data instead of DOM Blobs, which are the default.
+                ws.binaryType = "arraybuffer";
 
-            // If the server port isn't open, we get a close event
-            // at some point in the future (and will not get an onopen
-            // event)
-            ws.onclose = function () {
-                deferred.reject("WebSocket closed");
-            };
+                // If the server port isn't open, we get a close event
+                // at some point in the future (and will not get an onopen
+                // event)
+                ws.onclose = function () {
+                    ws = null;
+                    failure("WebSocket Closed");
+                };
 
-            ws.onopen = function () {
-                // If we successfully opened, remove the old onclose
-                // handler (which was present to detect failure to
-                // connect at all).
-                ws.onclose = null;
-                deferred.resolve([ws, port]);
-            };
-        });
+                // ws.onerror = function (e) {
+                //     singleAttemptDeferred.reject("Failed to connect to WebSocket: " + e.type);
```

Comment:
Note that I had *added* this as a test, and then forgot to remove it completely.  AFAICT, the onclose listener does what we need here